pdf/pdfannotation.cpp

```diff
@@ -0,0 +1,353 @@
+{
+    return d->m_annotation->author();
+}
+
+void PDFAnnotation::setAuthor(const QString &value)
+{
+    // Store last given author to use it as default for
+    // newly created annotations.
+    currentAuthor = value;
+
+    if (d->m_annotation->author() == value)
+        return;
+    d->m_annotation->setAuthor(value);
+    emit authorChanged();
+
+    d->m_annotation->setModificationDate(QDateTime());
+    emit modificationDateChanged();
+
+    if (d->m_document != nullptr)
+        d->m_document->documentModified();
+}
+
+QString PDFAnnotation::contents() const
+{
+    return d->m_annotation->contents();
+}
+
+void PDFAnnotation::setContents(const QString &value)
+{
+    if (d->m_annotation->contents() == value)
+        return;
+
+    d->m_annotation->setContents(value);
+    emit contentsChanged();
+
+    d->m_annotation->setModificationDate(QDateTime());
+    emit modificationDateChanged();
+
+    if (d->m_document != nullptr)
+        d->m_document->documentModified();
+}
+
+QDateTime PDFAnnotation::creationDate() const
+{
+    return d->m_annotation->creationDate();
+}
+
+QDateTime PDFAnnotation::modificationDate() const
+{
+    return d->m_annotation->modificationDate();
+}
+
+QColor PDFAnnotation::color() const
+{
+    return d->m_annotation->style().color();
+}
+
+void PDFAnnotation::setColor(const QColor &value)
+{
+    if (color() == value)
+        return;
+
+    Poppler::Annotation::Style style = d->m_annotation->style();
+    style.setColor(value);
+    d->m_annotation->setStyle(style);
+    emit colorChanged();
+
+    if (d->m_document != nullptr && d->m_page >= 0)
+        d->m_document->onPageChanged(d->m_page, d->m_annotation->boundary());
+
+    if (d->m_document != nullptr)
+        d->m_document->documentModified();
+}
+
+PDFAnnotation::SubType PDFAnnotation::type() const
+{
+    return (PDFAnnotation::SubType)d->m_annotation->subType();
+}
+
+
+static QStringList iconNames = (QStringList() << "Note" << "Comment" << "Key" << "Help" << "NewParagraph" << "Paragraph" << "Insert" << "Cross" << "Circle");
+
+PDFTextAnnotation::PDFTextAnnotation(QObject *parent)
+    : PDFAnnotation(new Poppler::TextAnnotation(Poppler::TextAnnotation::Linked), parent)
+{
+}
+
+PDFTextAnnotation::PDFTextAnnotation(Poppler::TextAnnotation *annotation,
+                                     PDFDocument *document, int ipage, QObject *parent)
+    : PDFAnnotation(annotation, document, ipage, parent)
+{
+}
+
+PDFTextAnnotation::~PDFTextAnnotation()
+{
+}
+
+void PDFTextAnnotation::attach(PDFDocument *document, PDFSelection *selection)
+{
+    if (!selection)
+        return;
+
+    QPair<int, QRectF> word = selection->rectAt(0);
+    attachAt(document, word.first, word.second.x(), word.second.y());
+}
+
+void PDFTextAnnotation::attachAt(PDFDocument *document,
+                                 unsigned int page, qreal x, qreal y)
+{
+    attachOnce(document, page);
+
+    d->m_annotation->setBoundary(QRectF(x, y, 24., 24.));
```

Comment:
Yes, I've explained this in one of the issue #80 comments. It is the size Poppler gives to the text annotation icon that is added in the document. It's also hard-coded in Poppler, and I didn't find any mean to retrieve this value.

I'll add a comment in our code about this.